src/components/use-view-switcher/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { useDispatch, select } from '@wordpress/data';
+import { Toolbar, ToolbarDropdownMenu } from '@wordpress/components';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { Icon, eye } from '../../icons';
+
+export const useViewSwitcher = ( clientId, views ) => {
+	const initialView = views[ 0 ];
+	const [ currentView, setCurrentView ] = useState( initialView );
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const { getBlock } = select( blockEditorStore );
+
+	const ViewSwitcherComponent = () => (
```

Comment:
We updated ViewSwticher since you copied this.
```suggestion
	const ViewSwitcherComponent => (
```